Specifying custom attributes on tags

mmiller42 opened the pull request:
Seems like a useful feature would be to add custom attributes to the script and link tags. For instance, it is common to want to add cross-origin, async, or subresource integrity attributes to a script tag.

jharris4 commented on spec/index.spec.js:
```diff
@@ -831,6 +863,129 @@ describe('HtmlWebpackIncludeAssetsPlugin', function () {
+      });
+    });
+
+    it('should include any files for a glob that does match files', function (done) {
```
How does this test relate to the new attributes feature?

jharris4 replied:
I'm just trying to follow what you were aiming to test here...

mmiller42 replied:
That would be me copying and pasting a little too much :)

jharris4 commented on index.js:
```diff
@@ -52,6 +52,22 @@ function isOneOf (v, values) {
 }
 
+function forOwn (object, iterator) {
+  for (var property in object) {
```
Just a style thing, but I usually use Object.keys and then iterating over the keys that way.

mmiller42 replied:
Okay, I can do that. I wasn't sure if you were aiming for maximum compatibility (for example, I didn't see any uses of `Array#forEach`), so I figured I would avoid `Object.keys()`. I'll update it.

jharris4 replied:
I just tend to use for loops because they're way faster according to many benchmarks. Probably premature optimization in most cases, but still nice to be consistent with code style if possible

jharris4 commented:
Also, interesting plugin you have going (https://github.com/mmiller42/html-webpack-externals-plugin). It does something similar to a plugin I was working on (https://github.com/jharris4/html-webpack-deploy-assets-plugin)

If `html-webpack-externals-plugin` does what I use `html-webpack-deploy-assets-plugin` for, I'd be happy to scrap my plugin and use yours... ;-)

mmiller42 commented:
Ha! I didn't know someone else had created a similar plugin.

Yeah, I think our plugins accomplish the same thing, ultimately. Your plugin seems more configurable in terms of where the copied files are outputted to, whereas my plugin also updates the base Webpack config to exclude modules from the bundle (so that `import React from 'react';` passes your linter but doesn't bundle React in, for example).

jharris4 commented:
Yeah. One problem I ran into was trying to use globs with assets. turns out `copy-webpack-plugin` doesn't actually copy the assets over until after `html-webpack-plugin` has already terminated, so I got a bit stuck there. Would be curious to know if you ran into a similar issue or had any workarounds...

jharris4 commented:
If you don't mind answering the question about what that one test was for (commented above), and adding an example to the README, I'll try and get this merged ASAP.

mmiller42 commented:
Okay, removed that pointless duplicated test and updated the readme.

To answer your question re: globs -- I don't think I have a better solution. All modules that are to be externalized must be explicitly specified; the only globs that my plugin uses are for copying over additional dependency files for a particular module (e.g. the fonts assets that Bootstrap needs to reference).

jharris4 commented:
Thanks! Released as `1.0.1`.

jharris4 commented:
Oh and regarding globs, yeah that's why I had to add the weird `globPath` option to make it search in a different location than the the webpack output folder.

Would've been nice to have better integration with the copy plugin, but maybe it'll be possible soon since it looks like the webpack contrib team have taken over ownership of it...
